server/webhook.go

```diff
@@ -722,105 +711,147 @@ func (s *Server) tryUpdateTasksFromModifiedFile(ctx context.Context, databases [
-		// Having no schema info indicates that the file is not a schema file (e.g.
-		// "*__LATEST.sql"), try to parse the migration info see if it is a data update.
-		migrationDetailList, activityCreateList = s.prepareIssueFromPushEventSDL(ctx, repo, pushEvent, schemaInfo, file, webhookEndpointID)
-		migrationDescription = "Apply_schema_diff"
-		migrationType = db.Migrate
-	} else {
-		// NOTE: We do not want to use filepath.Join here because we always need "/" as the path separator.
-		migrationInfo, err := db.ParseMigrationInfo(file, path.Join(repo.BaseDirectory, repo.FilePathTemplate))
+		// There are possibly multiple files in the push event.
+		// Each file corresponds to a (database name, schema version) pair.
+		// We want the migration statements are sorted by the file's schema version, and grouped by the database name.
+		fileListSorted, migrationInfoList, err := sortFilesBySchemaVersionGroupByDatabase(repo, nonSchemaFileList)
 		if err != nil {
-			log.Error("Failed to parse migration info",
-				zap.Int("project", repo.ProjectID),
-				zap.Any("pushEvent", pushEvent),
-				zap.String("file", file),
-				zap.Error(err),
-			)
-			return "", false, nil, nil
+			return "", false, nil, echo.NewHTTPError(http.StatusInternalServerError, "Failed to sort files by schema version group by database").SetInternal(err)
+		}
+		for i, file := range fileListSorted {
+			migrationDetailListForFile, activityCreateListForFile := s.prepareIssueFromPushEventDDL(ctx, repo, pushEvent, file.fileName, file.itemType, webhookEndpointID, migrationInfoList[i])
+			activityCreateList = append(activityCreateList, activityCreateListForFile...)
+			migrationDetailList = append(migrationDetailList, migrationDetailListForFile...)
+			// If there's one schema migration file, we set the issue type as MIGRATE.
+			if migrationInfoList[i].Type == db.Migrate {
+				migrationType = db.Migrate
+			}
+		}
+	} else {
+		var schemaFileList []distinctFileItem
+		var schemaInfoListForSDL []map[string]string
+		var nonSchemaFileList []distinctFileItem
+		for i, si := range schemaInfoList {
+			if si != nil {
+				schemaFileList = append(schemaFileList, filesToProcess[i])
+				schemaInfoListForSDL = append(schemaInfoListForSDL, si)
+			} else {
+				nonSchemaFileList = append(nonSchemaFileList, filesToProcess[i])
+			}
+		}
+		// If there's one schema state file, we set the issue type as MIGRATE.
+		if len(schemaFileList) != 0 {
+			migrationType = db.Migrate
 		}
-		migrationDescription = migrationInfo.Description
-		migrationType = migrationInfo.Type
 
-		if repo.Project.SchemaChangeType == api.ProjectSchemaChangeTypeSDL && migrationInfo.Type != db.Data {
-			activityCreate := getIgnoredFileActivityCreate(repo.ProjectID, pushEvent, file, errors.Errorf("Only DATA type migration scripts are allowed but got %q", migrationInfo.Type))
-			return "", false, []*api.ActivityCreate{activityCreate}, nil
+		// State based migration.
```

Comment:
For state-based migration, we can create separate issues for each database or one issue for a single group of tenant databases.

Reply:
The only difference between SDL v.s. DDL project will be who updates LATEST.sql file. We can ignore the path for SDL for now.

Reply:
For SDL, changed back to create issue per schema file.